### PR TITLE
Bp 0536 fix so exception

### PR DESCRIPTION
**Pull request by Johannes-Schneider**

### CODE REVIEW

File: src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java [hunk 1]
```diff
@@ -397,7 +399,31 @@ public int getEntityTypesCount() {
...
 
-		return databaseAccess.getArtifacts(session, query, transaction, query::getResultList, new ArrayList<>());
+		List<Event> events = databaseAccess.getArtifacts(session, query, transaction, query::getResultList, new ArrayList<>());
+
+		// we need to fetch more events, because there are still entityIds, which are not covered and we did not get enough events yet
+		if (events.size() < Math.abs(listEndIndex - listStartIndex) && limitedEntityIds.size() < entityIds.length) {
+			List<Long> entityIdsLeft = Arrays.asList(entityIds);
+			entityIdsLeft.removeAll(limitedEntityIds);
+
+			events.addAll(getEvents(eventTypeId, 0, Math.abs(listEndIndex - listStartIndex) - events.size(), entityIdsLeft.toArray(new Long[entityIdsLeft.size()])));
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java]: MAJOR Split this 156 characters long line (which is greater than 150 authorized). [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java [hunk 2]
```diff
@@ -397,7 +399,31 @@ public int getEntityTypesCount() {
...
+		}
+
+		events.sort(new Comparator<Event>() {
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java]: MAJOR Make this anonymous inner class a lambda [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1604)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java [hunk 3]
```diff
@@ -23,6 +23,8 @@
 
 import javax.persistence.Table;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java]: MINOR Remove this unused import 'java.util.Comparator'. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Review by julianweise — APPROVED** [review]
Das hast du sehr schön gemacht!

**Pull request merged**